apis/v1beta1/collector_webhook.go

```diff
@@ -290,6 +292,11 @@ func (c CollectorWebhook) Validate(ctx context.Context, r *OpenTelemetryCollecto
 		return warnings, fmt.Errorf("the OpenTelemetry Collector mode is set to %s, which does not support the attribute 'deploymentUpdateStrategy'", r.Spec.Mode)
 	}
 
+	components := r.Spec.Config.GetEnabledComponents()
+	if notAllowedComponents := c.fips.DisabledComponents(components[KindReceiver], components[KindExporter], components[KindProcessor], components[KindExtension]); notAllowedComponents != nil {
```

Comment:
given fips is an interface we should check if its nil prior to calling this. Doing it this way will also mean we can remove the noop fips check